src/textual/widgets/_digit_display.py

```diff
@@ -0,0 +1,337 @@
+┣━┓
+┗━┛
+"""
+
+_character_map[
+    "7"
+] = """
+╺━┓
+  ╹
+ ╹·
+"""
+
+_character_map[
+    "8"
+] = """
+┏━┓
+┣━┫
+┗━┛
+"""
+
+_character_map[
+    "9"
+] = """
+┏━┓
+┗━┫
+╺━┛
+"""
+
+_character_map[
+    " "
+] = """
+···
+···
+···
+"""
+
+
+_character_map[
+    "X"
+] = """
+╻ ╻
+·╋·
+╹ ╹
+"""
+
+_character_map[
+    "Y"
+] = """
+╻ ╻
+·┳·
+·╹·
+"""
+
+_character_map[
+    "Z"
+] = """
+╺━┓
+·▞·
+┗━╸
+"""
+
+
+_character_map[
+    "."
+] = """
+··
+··
+·•
+"""
+
+
+_character_map[
+    ","
+] = """
+··
+··
+·▞
+"""
+
+_character_map[
+    "+"
+] = """
+···
+╺╋╸
+···
+"""
+
+_character_map[
+    "-"
+] = """
+···
+╺━╸
+···
+"""
+
+_character_map[
+    "="
+] = """
+··
+╺━
+╺━
+"""
+
+_character_map[
+    "*"
+] = """
+···
+·✱·
+···
+"""
+
+
+_character_map[
+    "/"
+] = """
+··╻
+·▞·
+╹··
+"""
+
+_character_map[
+    "^"
+] = """
+·╻·
+▝·▘
+···
+"""
+
+
+# here we strip spaces and replace virtual spaces with spaces
+_character_map = {
+    k: v.strip().replace(_VIRTUAL_SPACE, " ") for k, v in _character_map.items()
+}
+
+
+class _SingleDigitDisplay(Static):
+    """
+    A widget to display a single digit or basic arithmetic symbol using Unicode blocks.
+    """
+
+    digit = reactive(" ", layout=True)
+    """The digit to display."""
+
+    DEFAULT_CSS = """
+    _SingleDigitDisplay {
+      height: 3;
+      min-width: 2;
+      max-width: 3;
+    }
+    """
+
+    def __init__(
+        self,
+        initial_value: str = " ",
+        expand: bool = False,
+        shrink: bool = False,
+        markup: bool = True,
+        name: str | None = None,
+        id: str | None = None,
+        classes: str | None = None,
+        disabled: bool = False,
+    ):
+        """
+        Create a single digit display widget.
+
+        Example:
+            ```py
+            class Example(App):
+                def compose(self) -> ComposeResult:
+                    return _SingleDigitDisplay("1")
+        """
+        super().__init__(
+            expand=expand,
+            shrink=shrink,
+            markup=markup,
+            name=name,
+            id=id,
+            classes=classes,
+            disabled=disabled,
+        )
+        self.digit = initial_value
+        self._content_width = 3
+
+    def validate_digit(self, digit: str) -> str:
+        """Sanitize and validate the digit input."""
+        if len(digit) > 1:
+            raise ValueError(f"Expected a single character, got {len(digit)}")
+        digit = digit.upper()
+        if digit not in _character_map:
+            raise ValueError(f"Unsupported character: {digit}")
+        return digit
+
+    def _watch_digit(self, digit: str) -> None:
+        """Called when the digit attribute changes and passes validation."""
+        content = _character_map[digit.upper()]
+        self._content_width = len(content.splitlines()[0])
+        self.update(content)
+
+    def get_content_width(self, container: Size, viewport: Size) -> int:
+        return self._content_width
```

Comment:
Are all the digits not the same size? If so, could we just return 3 here?

Reply:
I was trying to make the dot and comma to change width to 2, but in the end it's simpler to have every digit the same size indeed :+1: 
Lemme update this!

Reply:
updated!